### PR TITLE
Add citation.cff file

### LINKED ISSUE #2848: Add citation entry for the GitHub repository
Including a CITATION.cff file in the NetCDF repository would provide quick and clear guidelines on properly citing the library. This would be especially helpful for users looking to reference NetCDF in their research. The CITATION.cff file can be easily understood by GitHub and displayed on the repository sidebar.

### PR DESCRIPTION
**Pull request by WardF**
* Fixes #2848 

Initial pass, will be fleshed out with feedback from appropriate folks @ Unidata.

### CODE REVIEW

**Review by DennisHeimbigner — DISMISSED** [review]

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Visit https://bit.ly/cffinit to generate yours today!
+
+cff-version: 1.2.0
+title: 'NetCDF-C '
+message: >-
+  If you use this software, please cite it using the
+  metadata from this file.
+type: software
+authors:
+  - name: NSF Unidata
```
Comment (edwardhartnett) [inline CITATION.cff:11]: I would like it if we could list the authors, in addition to saying NFS Unidata.

Comment (WardF) [reply CITATION.cff:11]: That is the 'waiting to hear back' part referenced above, we will be listing the authors in the final version :).

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    family-names: Rew
+    affiliation: UCAR/Unidata
+  - given-names: Dennis
+    family-names: Heimbigner
+    affiliation: UCAR/Unidata
+  - given-names: Edward
+    family-names: Hartnett
+    affiliation: 'UCAR/Unidata, NOAA'
+  - given-names: Ward
+    family-names: Fisher
```
Comment (edwardhartnett) [inline CITATION.cff:23]: Glenn Davis also should be listed.

I suggest you take the list of names from the user guide in that order.

**Comment by WardF** [comment]
Putting a pin in this until next week, other bigger fish to fry and I need clarification on citations that will have to wait until after AMS.

File: CITATION.cff (hunk 2 above)
Comment (edwardhartnett) [reply CITATION.cff:26]: OK never mind I see he is there. ;-)

Comment (WardF) [reply CITATION.cff:26]: No problem; I said I was putting a pin in it so of course I then went and immediately took care of it so I could cross it off and focus on the other dozen pinned items XD

**Pull request merged**